data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/configuration/EntryConfig.java

```diff
@@ -22,6 +22,9 @@ public class EntryConfig {
     @JsonProperty("fields")
     private List<String> fields;
 
+    @JsonProperty("geoip_when")
```

Comment:
What is the motivation for this change? Why do we connect the when to each entry?

Reply:
This gives more flexibility to users to have when condition on each entry that at processor level. If user wants to have different condition on each entry they can do it in same processor instance than creating new instance.

Reply:
All our other processors do this in the top-level. And if we move toward a processor-level `when`, that would apply there. I tend to think we should support consistency over this.

Users who need the level of complexity where they are utilizing multiple conditions could run multiple processors as alternative. I think we should take that approach instead.

Reply:
The mutate event processors with entries have `when` condition on the `entry` level.